test/hotspot/jtreg/compiler/rangechecks/TestArrayAccessAboveRCAfterSinking.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2024, Red Hat, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8323274
+ * @summary sinking an array load out of loop can cause it to become dependent on a test other than its range check
+ * @run main/othervm -XX:-UseOnStackReplacement -XX:-TieredCompilation -XX:-BackgroundCompilation TestArrayAccessAboveRCAfterSinking
+ */
+
+
+import java.util.Arrays;
+
+public class TestArrayAccessAboveRCAfterSinking {
+    public static void main(String[] args) {
+        boolean[] allFalse = new boolean[100];
+        boolean[] allTrue = new boolean[100];
+        Arrays.fill(allTrue, true);
+        int[] array = new int[100];
+        for (int i = 0; i < 20_000; i++) {
+            test1(allTrue, array, 0, true, 0);
+            test1(allTrue, array, 0, false, 0);
+            inlined1(allFalse, array, 2, 0);
+            inlined1(allFalse, array, 42, 0);
+            inlined1(allTrue, array, 2, 0);
+            test2(allTrue, array, 0, true, 0);
+            test2(allTrue, array, 0, false, 0);
+            inlined2(allFalse, array, 2, 0);
+            inlined2(allFalse, array, 42, 0);
+            inlined2(allTrue, array, 2, 0);
+        }
+        try {
+            test1(allTrue, array, -1, true, 0);
+            test2(allTrue, array, -1, true, 0);
+        } catch (ArrayIndexOutOfBoundsException arrayIndexOutOfBoundsException) {
```

Comment:
Is there a reason both are together in the same `try-catch` block? It seems like if the first throws, then the second is never executed (and may as well be removed), or we only expect the second to throw, and so the first one should be outside the bock. What do you think?

Reply:
You're right. I will change that.